src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/PostgresAccessPolicyDAO.scala

```diff
@@ -307,9 +307,28 @@ class PostgresAccessPolicyDAO(protected val dbRef: DbReference,
     val ra = ResourceActionTable.syntax("ra")
     val paCol = PolicyActionTable.column
     if (actions.nonEmpty) {
-      samsql"""insert into ${PolicyActionTable.table} (${paCol.resourcePolicyId}, ${paCol.resourceActionId})
+      val insertQuery = samsqls"""insert into ${PolicyActionTable.table} (${paCol.resourcePolicyId}, ${paCol.resourceActionId})
             select ${policyId}, ${ra.result.id} from ${ResourceActionTable as ra} where ${ra.action} in (${actions})"""
-        .update().apply()
+
+      val inserted = samsql"$insertQuery".update().apply()
+
+      if (inserted != actions.size) {
+        // in this case some actions that we want to insert did not exist in ResourceActionTable
+        // add them now and rerun the insert ignoring conflicts
+        // this case should happen rarely
+        import SamTypeBinders._
+        val r = ResourceTable.syntax("r")
+        val p = PolicyTable.syntax("p")
+        val resourceTypePK = samsql"select ${r.result.resourceTypeId} from ${PolicyTable as p} join ${ResourceTable as r} on ${p.resourceId} = ${r.id}"
```

Comment:
Add a where clause to select the `resourceTypePK` for just this policy: 
```suggestion
        val resourceTypePK = samsql"select ${r.result.resourceTypeId} from ${PolicyTable as p} join ${ResourceTable as r} on ${p.resourceId} = ${r.id} where ${p.id} = ${policyId}"
```

Reply:
good catch